flink-queryable-state/flink-queryable-state-runtime/src/main/java/org/apache/flink/queryablestate/client/proxy/KvStateClientProxyHandler.java

```diff
@@ -225,7 +225,9 @@ private void executeActionAsync(
 
 			return location;
 		} else {
-			return FutureUtils.completedExceptionally(new UnknownLocationException("Could not contact the state location oracle to retrieve the state location."));
+			return FutureUtils.completedExceptionally(
+				new UnknownLocationException("Could not contact the state location oracle to retrieve the state location for state="
+					+ queryableStateName + " of job=" + jobId + ", the caused reason maybe the state is not ready or there is no job exists."));
```

Comment:
This message is pretty verbose and it contains grammatical errors. If it were to write this, it should be something like:

```
Could not contact the state location oracle to retrieve the location for state=QSName of job= JobID. The reason can be that the state is not ready or that that does not exist.
```

But before putting this in, it would be helpful if @florianschmidt1994 commented on what he thinks, as he is the one that opened the issue.

Reply:
I think this is a very good step in the right direction! What I've been wondering is 

a. Maybe we could look into whether or not we could distinguish the two cases and have separate error messages (this would require some more effort and I don't know if this is easily doable)

b. Do we need to expose the term "state location oracle" here? As an outstanding person I am not familiar with it and I think it might more more of a confusion than helpful to someone encountering that error message, seeing as this is not a really exceptional case but more of a setup / timing thing